pkg/util/labels.go

```diff
@@ -18,8 +18,9 @@ const (
 	ErrorOnDifferentDstKeyValue
 )
 
-// AddObjectLabels adds new label(s) to a single runtime.Object
-func AddObjectLabels(obj runtime.Object, labels labels.Set) error {
+// AddObjectLabelsWithFlags will set labels on the target object, label overwrite behavior
+// is controlled by the flags argument.
+func AddObjectLabelsWithFlags(obj runtime.Object, labels labels.Set, flags int) error {
```

Comment:
You also need to pass the flags argument to addDeploymentConfigNestedLabels below this method.